### DISCUSSION TITLE
update M dynamically

### DISCUSSION THREAD

**Discussion by Tang115** [CURENT/andes#416] [General]
Dear Dr. Cui,

I would like to know if there is a way to update the machine start up time (M=2H) dynamically. For example, if I set up the simulation time to 10s and I want to update the M of a generator at 5s. I have tried using alter(), but there is no impact on the system dynamic. Could you give me some suggestions on this question?

Best

**Comment by cuihantao** [comment]
It’s supposed to work. Could you check ss.GENROU.M.v before and after the alteration? 

If the same, is this using a snapshot?
